app.rb

```diff
@@ -0,0 +1,98 @@
+require "sinatra/base"
+require "sinatra/reloader"
+require_relative "lib/database_connection"
+require_relative "lib/peep_repository"
+require_relative "lib/peep"
+require_relative "lib/user_repository"
+require_relative "lib/user"
+require "date"
+
+DatabaseConnection.connect
+
+class Application < Sinatra::Base
+  enable :sessions
+
+  configure :development do
+    
+    register Sinatra::Reloader
+    also_reload "lib/peep.rb"
+    also_reload "lib/peep_repository.rb"
+    also_reload "lib/user.rb"
+    also_reload "lib/user_repository.rb"
+  end
+
+  get "/" do
+    erb(:index)
+  end
+
+  get '/feed' do
+    @peeps = PeepRepository.new.all.reverse
+    @user_repo = UserRepository.new
+    erb(:feed)
+  end
+
+  get '/sign_up' do
+    erb(:sign_up)
+  end
+
+   #create a new user
+  post "/users" do
+    repo = UserRepository.new
+    new_user = User.new
+    new_user.name = params[:name]
+    new_user.username = params[:username]
+    new_user.email = params[:email]
+    new_user.password = params[:password]
+    repo.create(new_user)
+    return erb(:user_created)
+  end
+
+  get '/login' do
+    erb(:login)
+  end
+
+  post '/login' do
+    email = params[:email]
+    password = params[:password]
+
+    users = []
+    user_repo = UserRepository.new
+    users << user_repo.find_by_email(email)
```

Comment:
If you only need access to the one user with the given email address (the user currently trying to sign in), maybe you don't need an array of `users`?